backend/src/modules/users/services/get.user.service.ts

```diff
@@ -38,4 +45,44 @@ export default class GetUserServiceImpl implements GetUserService {
 	getAllUsers() {
 		return this.userModel.find().select('-password -currentHashedRefreshToken').lean().exec();
 	}
+
+	async getAllUsersWithTeams() {
+		const users = await this.getAllUsers();
+		const results: UserWithTeams[] = [];
+
+		const usersOnlyWithTeams = await this.getTeamService.getUsersOnlyWithTeams();
+
+		users.forEach((user) => {
+			let userHasTeam = false;
+			const teams: string[] = [];
+			usersOnlyWithTeams.forEach((userOnlyWithTeams) => {
+				if (user._id.toString() === userOnlyWithTeams._id.toString()) {
+					userHasTeam = true;
+
+					const { teamsNames } = userOnlyWithTeams;
+
+					teamsNames.forEach((element) => {
+						element.forEach((el) => {
+							teams.push(el);
+						});
+					});
+				}
+			});
+
+			if (userHasTeam) {
+				const userWithTeams: UserWithTeams = {
+					user,
+					teams
+				};
+				results.push(userWithTeams);
+			} else {
+				const userWithTeams: UserWithTeams = {
+					user
+				};
+				results.push(userWithTeams);
+			}
+		});
+
+		return results;
```

Comment:
```suggestion
                const users = await this.getAllUsers();
		const mappedUsers: UserWithTeams[] = users.map((userFound) => {
			return {
				user: userFound as UserDto,
				teams: []
			};
		});

		const usersOnlyWithTeams = await this.getTeamService.getUsersOnlyWithTeams();

		const ids = new Set(usersOnlyWithTeams.map((userWithTeams) => String(userWithTeams.user._id)));

		return [
			...usersOnlyWithTeams,
			...mappedUsers.filter((userWithTeams) => !ids.has(String(userWithTeams.user._id)))
		];
```
This will merge both arrays but only push a user without a team if the `userOnlyWithTeams` array doesn't already have that user